src/hotspot/share/c1/c1_InstructionPrinter.cpp

```diff
@@ -861,6 +861,15 @@ void InstructionPrinter::do_RangeCheckPredicate(RangeCheckPredicate* x) {
   }
 }
 
+void InstructionPrinter::do_Blackhole(Blackhole* x) {
+  if (x->v() != NULL) {
```

Comment:
Should be an assert instead? (LIRGenerator::do_Blackhole doesn't expect NULL.)

Reply:
Would be sad to `assert` in essentially debugging code. I see that `print_value` does the right thing: prints `NULL`, if any. So I simplified this piece of code to call `print_value` unconditionally.